recipes/sbp/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+import os
+
+from textwrap import dedent
+
+import yaml
+
+from conans import ConanFile, CMake, tools
+
+
+class SbpConan(ConanFile):
+    name = "sbp"
+    license = "MIT"
+    author = "Swift Navigation <dev@swiftnav.com>"
+    url = "https://github.com/swift-nav/libsbp"
+    description = "Swift Binary Protocol client library"
+    topics = ("gnss",)
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": True, "fPIC": True}
+    generators = "cmake"
+    generators = "cmake"
+    exports = "submoduledata.yml"
+    exports_sources = "c"
+
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder
+        )
+
+        submodule_filename = os.path.join(self.recipe_folder, "submoduledata.yml")
+        with open(submodule_filename, "r") as submodule_stream:
+            submodules_data = yaml.load(submodule_stream)
+            for path, submodule in submodules_data["submodules"][self.version].items():
+                submodule_data = {
+                    "url": submodule["url"],
+                    "sha256": submodule["sha256"],
+                    "destination": os.path.join(self._source_subfolder, path),
+                    "strip_root": True,
+                }
+
+                tools.get(**submodule_data)
+
+        tools.replace_in_file(
```

Comment:
Please move the patching to `build`.

Reply:
note this was copied from the example at https://docs.conan.io/en/latest/creating_packages/getting_started.html#creating-the-package-recipe which puts this in the `.source`.  Should that be updated?

Reply:
It's a convention we're using at conan-center-index.
The conan documentation should be treated as just that: the documentation of conan, not the conventions of center index.

Not that patching in `build` is more flexible, as it allows to conditonally patch sources depending on options/settings.